[author][new] Add can w/ npm auto-update via single package.json

christopherjbaker opened the pull request:
Adding CanSJ 3 [npm:can](https://www.npmjs.com/package/can).

We were previously added to your site under [can.js](https://cdnjs.com/libraries/can.js) (we used to be named [npm:canjs](https://www.npmjs.com/package/canjs)), though it has not been updated in quite some time (and those files are actually from [can-jquery](https://www.npmjs.com/package/can-jquery). Should something be done about the now incorrectly named can.js?


Checklist for **Pull request** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/canjs/canjs
 * Official website (optional, not the repository): http://canjs.com/
 * NPM package url (optional): https://www.npmjs.com/package/can
 * License and its reference: [MIT](https://github.com/canjs/canjs/blob/master/license.md)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 134
   - Count of stars: 1476
   - Count of forks: 375
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 1110
   - Downloads in the last month: 9066

# Essential checklist
 * [X] I'm the author of this library **(one of them)**
   * [X] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 500 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

christopherjbaker commented:
I made those changes. At this point, however, I have only edited the file in the browser, and haven't worked with such large repos before. Would it be too much trouble to squash it for me?

PeterDaveHello commented:
@christopherjbaker we can squash for you, no worries.

pvnr0082t commented on ajax/libs/can/package.json:
```diff
@@ -0,0 +1,31 @@
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:canjs/canjs.git",
+    "github": "https://github.com/canjs/canjs"
```
We don't need github attribute. Please remove it. Thanks.

pvnr0082t commented on ajax/libs/can/package.json:
```diff
@@ -0,0 +1,31 @@
+  "name": "can",
+  "title": "CanJS",
+  "description": "MIT-licensed, client-side, JavaScript framework that makes building rich web applications easy.",
+  "filename": "can.js",
```
"filename" should point to the main minified file. I think it would be `"can.min.js"` here.

pvnr0082t commented on ajax/libs/can/package.json:
```diff
@@ -0,0 +1,31 @@
+    "github": "https://github.com/canjs/canjs"
+  },
+  "license": "MIT",
+  "npmName": "can",
```
There are 138 releases on GitHub but 111 on NPM. It would be better to use git auto-update config instead. How do you think @christopherjbaker? You can take a look at [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md).

christopherjbaker replied:
I could go either way on this. Since NPM is the official distribution mechanism, that made more sense to me. The github repo also does not include the built files, so I'm not sure we could use the git auto-update anyway.

pvnr0082t replied:
Okay. Let's keep using npm auto-update config. :+1:

pvnr0082t commented:
ping @maruilian11

PeterDaveHello commented:
Look like we already got it here: https://cdnjs.com/libraries/can.js, @christopherjbaker maybe we should just update the origin one?

christopherjbaker commented:
I mentioned that in the initial pull. We have since changed our npm name from `canjs` to just `can`, so I thought it more correct to name the folder after the repo. It can stay if that is preferable to you, though.

PeterDaveHello commented:
@christopherjbaker sorry that I may forgot that, I think it's better to keep using the origin one, just update its package.json to make the updates in, there are some cases on cdnjs using this method, especially the change isn't that huge, thanks!

christopherjbaker commented:
I removed can and updated can.js. CI is alerting that can/package.json cannot be found (perhaps because of the branch name?). Should I make a new branch/PR with that update, or is there a way to fix this?

PeterDaveHello commented:
@christopherjbaker since the commits are not squashed, so the changes of can are still there in the history, I'll squash it for you, give me 5 mins please, and then we can see if it can pass the test!

PeterDaveHello commented:
Thanks @christopherjbaker
